tensorboard/plugins/mesh/README.md

```diff
@@ -0,0 +1,111 @@
+# Mesh Plugin
+
+## Overview
+Mesh and point cloud are important and powerful types of data to represent 3D
+shapes and widely studied in the field of computer vision and computer 
+graphics.
+3D data is becoming more ubiquitous and researchers challenge new problems like
+3D geometry reconstruction from 2D data, 3D point cloud semantic segmentation,
+aligning or morphing 3D objects and so on. Therefore, visualizing results,
+especially during the training stage, is critical to better understand how
+the model performs.
+
+![Mesh Plugin in TensorBoard](
+https://storage.googleapis.com/tensorflow-graphics/git/readme/tensorboard_plugin.jpg)
+
+This plugin intends to display 3D point cloud or mesh (triangulated point 
+cloud)
+in TensorBoard. In addition, it allows the user to
+interact with the rendered object.
+
+## Summary API
+Either a mesh or a point cloud can be represented by a set of tensors. For
+example, one can viewpoint cloud as a set of 3D coordinates of the points and
+some colors associated with each point.
+
+```python
+from tensorboard.plugins import mesh
+...
+point_cloud = tf.constant([[[0.19, 0.78, 0.02], ...]], shape=[1, 1064, 3])
+point_colors = tf.constant([[[128, 104, 227], ...]], shape=[1, 1064, 3])
+
+summary = mesh.summary('point_cloud', vertices=point_cloud, colors=point_colors)
+```
+
+NOTE `colors` tensor is optional in this case but can be useful to show
+different semantics of the points.
+
+The plugin currently supports only triangular meshes which are different from
+point clouds above only by the presence of faces - set of vertices representing
+the triangle on the mesh.
+
+```python
+mesh = tf.constant([[[0.19, 0.78, 0.02], ...]], shape=[1, 1064, 3])
+colors = tf.constant([[[128, 104, 227], ...]], shape=[1, 1064, 3])
+faces = tf.constant([[[13, 78, 54], ...]], shape=[1, 752, 3])
+
+summary = mesh.summary('mesh', vertices=mesh, colors=colors, faces=faces)
+```
+
+Again, only the `colors` tensor is optional for mesh summaries.
+
+## Scene configuration
+Another aspect of displaying 3D visualizations is a scene configuration, i.e.
+intensity and color of light sources, objects' material, camera models and so
+on. All of that can be configured via an additional parameter `config_dict`.
+This dictionary may contain three high-level keys: `camera`, `lights` and
+`material`. Each key must also be a dictionary with mandatory key `cls`,
+representing valid [THREE.js](https://threejs.org) class name.
+
+```python
+camera_config = {'cls': 'PerspectiveCamera'}
+summary = mesh.summary('mesh', vertices=mesh, colors=colors, faces=faces,
+  config_dict={'camera': camera_config})
+```
+
+`camera_config` from the snippet above can be expanded according to the
```

Comment:
what do you mean that it can be “expanded”? I don’t understand what this
sentence means.

Inferring from context, perhaps these paragraphs would be clearer as

``````markdown
The `cls` key of the `camera_config` dict in the snippet above
corresponds to the name of a THREE.js class, and all other keys in the
dict will be passed to that class’s constructor. For example, referring
to the [`PerspectiveCamera` documentation][pc-doc], we might write:

```python
camera_config = {
  'cls': 'PerspectiveCamera',
  'fov': 0.75,
  'aspect': 0.9,
}
```

Consult the [THREE.js documentation][three-doc] for more details.
``````

assuming that I’m understanding correctly?

[three-doc]: https://threejs.org/docs/index.html#manual/en/introduction/Creating-a-scene
[pc-doc]: https://threejs.org/docs/index.html#api/en/cameras/PerspectiveCamera

Reply:
this works. What I meant was "you can add keys here if they correspond to something in the docs"